cli/Sources/TuistServer/Services/MultipartUploadArtifactService.swift

```diff
@@ -100,18 +103,20 @@
                         let currentPartNumber = partNumber.value
                         partNumber.mutate { $0 += 1 }
                         group.addTask {
-                            let uploadURLString = try await generateUploadURL(MultipartUploadArtifactPart(
-                                number: currentPartNumber,
-                                contentLength: bytesRead
-                            ))
-                            guard let url = URL(string: uploadURLString) else {
-                                throw MultipartUploadArtifactServiceError.invalidMultipartUploadURL(uploadURLString)
+                            try await retryProvider.runWithRetries {
```

Comment:
I'm adding an initial retry for the parts – right now, if a part fails, we retry the _whole_ multipart upload, which is inefficient to say the least